renderer/plugin.js

```diff
@@ -192,15 +201,47 @@ org.ekstep.contentrenderer.questionUnitPlugin = Plugin.extend({
    * @param {{src:String}} assetObj
    * @example toggleAudio(src: "/assets/public/content/rani1_1466755651199.mp3")
    */
-  toggleAudio: function(assetObj) {
+  toggleAudio: function (assetObj) {
     HTMLAudioPlayer.togglePlay(this.getAssetUrl(assetObj.src));
   },
+  /**
+   * Invokes getIcon function, a adaptor for question unit components
+   * @memberof org.ekstep.questionunit
+   * @param {target:String} eventData
+   */
+  handleLoadImageFromUrl: function (eventData) {
+    var src = this.getIcon(eventData.target.path, eventData.target.pluginId, eventData.target.pluginVer);
+    eventData.target.element.attr('src', src);
+  },
+  handleGetAssetUrl: function(eventData){
+    var src = this.getAssetUrl(eventData.target.path, eventData.target.pluginId, eventData.target.pluginVer);
+    eventData.target.element.attr('src', src);
+  },
+  /**
+   * returns icon url
+   * @memberof org.ekstep.questionunit
+   * @param String eventData
+   * getIcon('renderer/assets/icon.png')
```

Comment:
getIcon & getAudioIcon can be combined to a single function
Similarly handleLoadImageFromUrl & handleGetAssetUrl can be combined.